datasource/opennms/src/main/java/org/opennms/oce/datasource/opennms/processors/InventoryTableProcessor.java

```diff
@@ -130,13 +157,34 @@ public void process(String key, InventoryModelProtos.InventoryObjects inventory)
             }
         } else {
             this.kvStore.put(key, inventory);
-            inventoryHandlers.forEach(h -> {
-                try {
-                    h.onInventoryAdded(toInventory(Collections.singletonList(inventory)));
-                } catch (Exception e) {
-                    LOG.error("onInventoryAdded() call failed with inventory: {} on handler: {}", inventory, h, e);
-                }
-            });
+
+            // Only handle inventory that has not been referenced before
+            List<InventoryObject> newInventory = toInventory(Collections.singletonList(inventory)).stream()
+                    .filter(io -> {
+                        Integer references = inventoryReferences.putIfAbsent(io, 1);
```

Comment:
On cold start the backing KeyValueStore may already contain values - `#process()` will not be called for these.

In this case there are already "references" to the object, so we cannot assume that the value only starts at 1 when the object is first seen. We'll also need to way to initialize the reference count when the processor is initialized.

Reply:
I've added synchronization between the ```#init()``` and ```#process()``` that I think addresses this now. Between those two methods, the first one to see the inventory get referenced should be able to safely initialize the reference count to 1.

I wasn't sure if Kafka already provides an ordering guarantee between init and process. I added the latch to handle that and it will be harmless if there is already ordering enforced.